acme/api/handler.go

```diff
@@ -149,31 +148,63 @@ func (h *Handler) GetAuthz(w http.ResponseWriter, r *http.Request) {
 	api.JSON(w, authz)
 }
 
-// GetChallenge ACME api for retrieving a Challenge.
```

Comment:
Gometalinter expects exported method/function documentation to start with the name of the method -- hence `GetChallenge ...`. We can disable that check but I think it's worthwhile to force us to document exported methods.

Reply:
FWIW the previous comment simply had "GetChallenge" prefixed onto an otherwise complete sentence (making it grammatically incorrect) in order to appease the linter:
```
// GetChallenge ACME api for retrieving a Challenge.
```
The coherent sentence is "ACME api for retrieving a Challenge", or perhaps "GetChallenge **is the** ACME api for retrieving a Challenge".  

So while I agree that we should document functions, if people are just going to smash the func name on the front of a sentence because the linter says so, I don't understand the value of the lint rule. Maybe it's so you land on the comment instead of the func name if you search? Or maybe it's to try an enforce a certain comment voicing? Is there a place I can read up on the lint rules?

Also, does the lint rule enforce a comment on all public functions? Or just enforce that if a comment exists, it should begin with the func name?

Reply:
> Also, does the lint rule enforce a comment on all public functions? Or just enforce that if a comment exists, it should begin with the func name?

It looks like the linter error message distinguishes between having a comment and the comment not being in the preferred format.

Reply:
This rule is also not as a custom configuration that we set up, it's the default and it is used all over the standard library and 3rd party libraries in Go, see for example https://golang.org/pkg/crypto/x509/#CreateCertificate

Reply:
I guess I was wondering why. Does golang have a blog post or anything regarding their preferred comment style, voicing, etc.? Just looking to inform myself and understand why the linter is asking for what it asks for.

Reply:
https://golang.org/doc/effective_go.html#commentary

Reply:
> Doc comments work best as complete sentences, which allow a wide variety of automated presentations. The first sentence should be a one-sentence summary that starts with the name being declared. 

Thanks!